ui/packages/catalog-ui-search/src/main/webapp/react-component/results-export/container.tsx

```diff
@@ -82,32 +81,27 @@ class ResultsExport extends React.Component<Props, State> {
     fetch(`./internal/transformers/${this.getTransformerType()}`)
       .then(response => response.json())
       .then((exportFormats: ExportFormat[]) => {
-        const resultsAttributes: any = {}
-        this.props.results.map((result: Result) => {
-          result.attributes.forEach((attr: any) => {
-            if (resultsAttributes[attr] === undefined) {
-              resultsAttributes[attr] = [result.id]
-            } else {
-              resultsAttributes[attr].push(result.id)
-            }
-          })
-        })
+        const attributesByResult = this.props.results.reduce(
+          (acc, result) => {
+            const attributes = new Set(result.attributes)
+            acc.push(attributes)
+            return acc
+          },
+          [] as Array<Set<string>>
+        )
         return exportFormats.filter(exportFormat => {
-          const requiredAttr = exportFormat['required-attributes']
-          if (!Array.isArray(requiredAttr) || requiredAttr.length == 0) {
-            return true
-          }
-          let resultIds = resultsAttributes[requiredAttr[0]]
-          if (resultIds && resultIds.length > 0) {
-            for (let index = 1; index < requiredAttr.length; index++) {
-              const currResultIds = resultsAttributes[requiredAttr[index]]
-              if (_.intersection(resultIds, currResultIds).length == 0) {
-                return false
-              }
-              resultIds = currResultIds
-            }
+          const requiredAttributes = exportFormat['required-attributes']
+          const attributesNotRequired =
+            !Array.isArray(requiredAttributes) ||
+            requiredAttributes.length === 0
+          if (attributesNotRequired) {
             return true
           }
+          return attributesByResult.some(resultAttributes =>
```

Comment:
Should it be `attributesByResult.every` instead of `attributesByResult.some`?

Reply:
I used `some` because the compressed export will work as long as at least one result has the attributes required by the chosen result transformer. The zip will contain only the export files for results that meet the requirements, because the export will always use the `ZipCompression` transformer (which delegates to the chosen result transformer). That transformer will produce a valid output even when some results can't be transformed using the chosen result transformer.

But now I'm wondering about the uncompressed case. In that case, we don't know exactly which transformer will be used, so we can't guarantee it will correctly handle results lacking the required attributes. Maybe the logic should be to use `some` for zipped exports, and `every` for unzipped exports.

Reply:
I've updated the logic as described.